### PR TITLE
Replace UserContext with redux state selectors

### PR DESCRIPTION
**Pull request by eikhr**
# Description

The UserContext was simply mirroring what was in the redux-state, and was completely unnecessary.

# Result

Only refactoring. I replaced all usage of `useUserContext` with either `useCurrentUser` or `useIsLoggedIn` which directly select from the redux state. I also did some minimal refactoring in the various components using `useCurrentUser`, because it now returns `undefined` instead of `{}` when the user is not yet loaded or not logged in.

Number of typescript errors reduced from 731 to 712:)

# Testing

- [x] I have thoroughly tested my changes.

I read through all my changes, and for all components where behaiviour might have changed, I did some manual testing of the component.

### CODE REVIEW

File: app/routes/events/components/EventDetail/index.tsx [hunk 1]
```diff
@@ -162,6 +164,8 @@ const EventDetail = () => {
       : selectPoolsForEvent(state, { eventId }),
   );
 
+  const showSkeleton = (fetching && isEmpty(event)) || !currentUser;
```
Comment (ivarnakken) [inline app/routes/events/components/EventDetail/index.tsx:167]: Won't this prevent unauthenticated users from viewing event details?

**Review by ivarnakken — APPROVED** [review]
This has bothered me for long! 🏅

File: app/routes/events/components/EventDetail/index.tsx (hunk 1 above)
Comment (eikhr) [reply app/routes/events/components/EventDetail/index.tsx:167]: Ahh, right! Good catch! That makes fixing the type-errors a bit harder...

Comment (eikhr) [reply app/routes/events/components/EventDetail/index.tsx]: I think this page will need to be refactored more later to fix all the types

**Pull request merged**